### PR TITLE
feat: config argv source, runtime port handling

### PR DESCRIPTION
**Pull request by joshgummersall**
Encapsulate port handling that has leaked into the generators ([see here](https://github.com/microsoft/botframework-components/pull/746/files#diff-4d0da3953d67d3a6198cbd063f0efcf70af9240a5a1e09875b50715d155ed5bfR7)).

Fixes https://github.com/microsoft/botbuilder-js/issues/3407

### CODE REVIEW

File: libraries/botbuilder-runtime/src/configuration.ts [hunk 1]
```diff
@@ -97,6 +108,31 @@ export class Configuration implements IConfiguration {
...
+        const runtype = Union(Number, String);
+        const value = await this.type(path, runtype);
+        if (value) {
+            return runtype.match(
+                (num) => num,
+                (str) => {
+                    const parsed = parseInt(str, 10);
+
+                    if (isNaN(parsed)) {
+                        throw new Error(`${str} is not an integer`);
```
Comment (stevengum) [inline libraries/botbuilder-runtime/src/configuration.ts:127]: How do you feel about throwing a `TypeError`? Do you have any stance on the SDK throwing the appropriate error vs the generic `Error`?

Comment (joshgummersall) [reply libraries/botbuilder-runtime/src/configuration.ts:127]: Yep - an oversight on my part! Will fix.

File: libraries/botbuilder-runtime/test/settings/base.json [hunk 2]
```diff
@@ -8,7 +8,11 @@
     "bad": false,
     "ok": "howdy"
   },
+  "numbers": {
+    "bad": "uh oh",
+    "ok": 10
+  },
```
Comment (stevengum) [inline libraries/botbuilder-runtime/test/settings/base.json:14]: Is the delta in base.json still necessary? The PR's initial test additions were removed.

Comment (joshgummersall) [reply libraries/botbuilder-runtime/test/settings/base.json:14]: Nope - good catch.

**Review by stevengum — APPROVED** [review]
LGTM 🚀

**Pull request merged**